### ISSUE TITLE
Document how to create .svg file

### ISSUE THREAD

**Issue by tacman** [lyrixx/SFLive-Paris2016-Workflow#9]
How is task/doc.svg.twig created?   When I dump the svg file:

    bin/console workflow:dump task | dot -Tsvg -o templates/task/task.svg

I get

![image](https://user-images.githubusercontent.com/619585/42125851-77206834-7c4c-11e8-9001-ac9e1c21ad6a.png)

Instead of what is in the repository in task/doc.svg.twig

![image](https://user-images.githubusercontent.com/619585/42125861-965e70b0-7c4c-11e8-813a-65a04ce35ee6.png)

I have Graphviz 2.4 installed.

What command is run to get the transitions in boxes, as you have it here?

**Comment by lyrixx** [comment]
Hi.

The command was broken since I refactored the app to use symfony flex structure.
It's fixed by #10 

Once merged, you will be able to build the SVG thanks to: `bin/console workflow:build:svg  workflow.article`

**Comment by tacman** [comment]
pulling master, then running bin/console workflow:build:svg workflow.article creates a different file than what's checked in.

**Comment by tacman** [comment]
The place name is no longer the same, so the javascript that highlights the current place no longer works.

```xml
<g id="edge3" class="edge">
<title>place_dba928f4ece84b8c361d19595284bfb3c79d03fe&#45;&gt;transition_e5fd9cfe0e8039111d54b588e77b2bb0cad41c3a</title>
```

```javascript
    $(function() {
        var name = "{{ task.marking }}".replace(/[^\w]/i, '_').toLowerCase();
        var $svg = $('#state_machine-task');

        $('g.node title:contains("place_'+name+'")', $svg).each(function(j, place) {
            var place = $(place).parent();
```

![image](https://user-images.githubusercontent.com/619585/42230106-2860068e-7eb6-11e8-979c-6002aa7346fd.png)

should be

![image](https://user-images.githubusercontent.com/619585/42230138-39d64306-7eb6-11e8-969f-82c030526fb1.png)

**Comment by lyrixx** [comment]
arf

**Comment by tacman** [comment]
@lyrixx Two approaches: revert the GraphViz dumper to use place names instead of the dotize() place name, or change the javascript code to look for the dotize'd placename.

Reverting seems easier and yields a more readable svg file, but presumably it was changed to use the hash for some reason.  

What do you suggest for the PR?

(Obviously, there are more solutions, like dynamically creating the dot file.  I'm working on a WorkflowExtensionBundle now that does that and a few more things, but it's still rough.)

**Comment by tacman** [comment]
OK, some success.  After spending way too long figuring out how to create sha1 hashes in javascript, I found a way to do it, but can't figure out how to make it a simple synchronous call.  It's async now and working for single state (e.g task), but not yet for article since I'm not sure how to handle the async calls to check each place.

I submitted a PR, but I'm not that familiar with the process (most PR's I submit are made right in the browser on github), so feedback on how to do that better is welcome.

**Comment by lyrixx** [comment]
Thanks for your PR :+1: I will look at it asap

**Comment by tacman** [comment]
I realized that I didn't check in the svg file.  In part because I don't
think it should be checked in, but rather generated as part of the setup
instructions.

I think that most users are going to download the demo, see it work, and
then tweak the workflow.yaml file and see how it changes.  I've used a
variation of this to configure my workflows, a psuedo-gui.  So change the
workflow, re-generate the svg file, refresh the page.
